### PR TITLE
Fixed various race conditions/errors in the tests

### LINKED ISSUE #2288: Running test on test_events_endpoint with other test using setUpFileServer produces error
Tested on the most updated `devel` branch.

I keep getting an error about `AssertionError: The reactor had readers/writers left` in some testcases. It seems the problem relies whenever I run **together** test : `test_events_endpoint` with other test that use `setUpFileServer` function. 

It is tested with `Test/test_libtorrent_download.py` and `Test/API/test_download.py`.

### LINKED ISSUE #2289: Error when running test on devel (due to DefaultDownloadStartupConfig singleton)
When running tests on devel, the following error is raised:

```
nosetests Tribler/Test/Core -xv
...
test_convert_default_download_config (Tribler.Test.Core.Upgrade.test_pickle_converter.TestPickleConverter) ... ERROR

======================================================================
ERROR: test_convert_default_download_config (Tribler.Test.Core.Upgrade.test_pickle_converter.TestPickleConverter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Test/test_as_server.py", line 81, in check
    result = fun(*argv, **kwargs)
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Test/Core/Upgrade/test_pickle_converter.py", line 43, in test_convert_default_download_config
    PickleConverter(self.mock_session).convert_default_download_config()
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Core/Upgrade/pickle_converter.py", line 136, in convert_default_download_config
    ddsconfig = DefaultDownloadStartupConfig()
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Core/DownloadConfig.py", line 225, in __init__
    raise RuntimeError("DefaultDownloadStartupConfig is singleton")
RuntimeError: DefaultDownloadStartupConfig is singleton
```

### LINKED ISSUE #2290: Error when running test on devel (due to overridden static method)
When running the unit tests locally, I get the following error:

```
nosetests Tribler/Test/Core/Modules/RestApi/test_my_channel_endpoints.py:TestMyChannelModifyTorrentsEndpoint.test_add_torrent_to_my_channel_500 Tribler/Test/Core/test_torrent_def.py -x
...
test_load_from_url (Tribler.Test.Core.test_torrent_def.TestTorrentDef) ... ERROR

======================================================================
ERROR: test_load_from_url (Tribler.Test.Core.test_torrent_def.TestTorrentDef)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Test/test_as_server.py", line 81, in check
    result = fun(*argv, **kwargs)
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Test/Core/test_torrent_def.py", line 204, in test_load_from_url
    torrent = TorrentDef.load_from_url("http://torrent.ubuntu.com/releases/wily/release/server/ubuntu-15.10-server-ppc64el.iso.torrent")
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Test/Core/Modules/RestApi/test_my_channel_endpoints.py", line 390, in fake_load_from_url
    raise ValueError("Test error")
ValueError: Test error
```

This is due to the fact that a static method (`TorrentDef.load_from_url`) is overriden by a 'fake' method. This has consequences for subsequent tests that are making use of the `load_from_url` method in `TorrentDef`: they all call that overriden method.

### LINKED ISSUE #2291: Error when running tests on devel (dirty reactor due to rss scrape task left)
When running the tests on devel, I got the following error:

```
nosetests Tribler/Test/Core/Modules/Channel/test_channel_rss.py Tribler/Test/Core/Modules/RestApi/test_my_channel_endpoints.py:TestMyChannelModifyTorrentsEndpoint -xv
...
Testing whether a torrent can be added to a channel using the API ... ERROR
======================================================================
ERROR: Testing whether a torrent can be added to a channel using the API
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Test/test_as_server.py", line 281, in tearDown
    super(TestAsServer, self).tearDown(annotate=False)
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Core/Utilities/twisted_thread.py", line 151, in errback
    failure.raiseException()
  File "/Library/Python/2.7/site-packages/twisted/internet/defer.py", line 588, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/Users/martijndevos/Documents/tribler_endpoints/Tribler/Test/test_as_server.py", line 151, in checkReactor
    self.assertFalse(delayed_calls, "The reactor was dirty when tearing down the test")
AssertionError: The reactor was dirty when tearing down the test
-------------------- >> begin captured logging << --------------------
```

Note the following delayed calls:

```
...
TestMyChannelModifyTorrentsEndpoint: DEBUG: Thread still running PoolThread-twisted.internet.reactor-0, daemon: True, instance: <Thread(PoolThread-twisted.internet.reactor-0, started daemon 123145310715904)>
twisted: INFO: (TCP Port 56969 Closed)
twisted: INFO: Stopping factory <twisted.web.server.Site instance at 0x1148d8248>
TestMyChannelModifyTorrentsEndpoint: ERROR: The reactor was dirty:
TestMyChannelModifyTorrentsEndpoint: ERROR: >     <DelayedCall 0x114570d88 [1794.27570295s] called=0 cancelled=0 ChannelRssParser._task_scrape()>
--------------------- >> end captured logging << ---------------------
```

Problem: the `ChannelRssParser` is not closed correctly in the `test_channel_rss` tests.

### PR DESCRIPTION
**Pull request by devos50**
Fixes #2289, fixes #2290, fixes #2291, fixes #2288.

### CODE REVIEW

**Comment by devos50** [comment]
`test_anon_download_without_relays (Tribler.Test.test_tunnel_community_negative.TestTunnelCommunityNegative)` timing out

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@lfdversluis can you review?

File: Tribler/Core/TorrentDef.py [hunk 1]
```diff
@@ -141,6 +141,7 @@ def load_from_url(url):
         """
         # Class method, no locking required
         try:
+            # TODO Martijn: this request should be done using Twisted
```
Comment (lfdversluis) [inline Tribler/Core/TorrentDef.py]: Create an issue for this :)

File: Tribler/Test/Core/Modules/Channel/test_channel_rss.py [hunk 2]
```diff
@@ -11,26 +11,35 @@ class TestChannelRss(BaseTestChannel):
 
     def setUp(self, annotate=True):
         """
-        Setup the tests by creating the ChannelRssParser instance.
+        Setup the tests by creating the ChannelRssParser instance and initializing it.
         """
         super(TestChannelRss, self).setUp(annotate=annotate)
         self.channel_rss = ChannelRssParser(self.fake_session, self.fake_channel_community, 'a')
+        self.channel_rss.initialize()
+        self.should_shutdown = True
```
Comment (lfdversluis) [inline Tribler/Test/Core/Modules/Channel/test_channel_rss.py]: You initialize stuff and want to shutdown on a setup?

File: Tribler/Core/TorrentDef.py (hunk 1 above)
Comment (devos50) [inline Tribler/Core/TorrentDef.py]: We already have: #2066

**Comment by lfdversluis** [comment]
Looks good.

File: Tribler/Test/Core/Modules/Channel/test_channel_rss.py (hunk 2 above)
Comment (devos50) [inline Tribler/Test/Core/Modules/Channel/test_channel_rss.py]: The `should_shutdown` variable is used to indicate whether we should invoke the shutdown method of `self.channel_rss` after the tests are done. Settings this variable to true does not immediately shutdown the instance.

Comment (lfdversluis) [inline Tribler/Test/Core/Modules/Channel/test_channel_rss.py]: Confusing imo

Comment (devos50) [inline Tribler/Test/Core/Modules/Channel/test_channel_rss.py]: Ok I will find a way to refactor it then.

**Comment by devos50** [comment]
@whirm processed feedback, can you take a look?

**Comment by whirm** [comment]
Meh, tunnel test got stuck again

**Comment by whirm** [comment]
retest this please

**Pull request merged**